src/test/java/unit/DataCollectionReferenced.java

```diff
@@ -0,0 +1,212 @@
+            "resolve", "--name=" + DATA_COLLECTION_RECOLLECTION_NAME);
+    assertDataCollectionResourceResolve(resolve);
+
+    // `terra resolve --name=$DATA_COLLECTION_RECOLLECTION_NAME/`
+    JSONObject resolve2 =
+        TestCommand.runAndGetJsonObjectExpectSuccess(
+            "resolve", "--name=" + DATA_COLLECTION_RECOLLECTION_NAME + "/");
+    assertDataCollectionResourceResolve(resolve2);
+
+    // `terra resolve
+    // --name=$DATA_COLLECTION_RECOLLECTION_NAME/$DATA_COLLECTION_BUCKET_RECOLLECTION_NAME`
+    JSONObject resolve3 =
+        TestCommand.runAndGetJsonObjectExpectSuccess(
+            "resolve",
+            "--name="
+                + DATA_COLLECTION_RECOLLECTION_NAME
+                + "/"
+                + DATA_COLLECTION_BUCKET_RECOLLECTION_NAME);
+    assertEquals(
+        ExternalGCSBuckets.getGsPath(DATA_COLLECTION_BUCKET_NAME),
+        resolve3.get(DATA_COLLECTION_BUCKET_RECOLLECTION_NAME));
+    assertEquals(1, resolve3.length());
+
+    // `terra resolve
+    // --name=$DATA_COLLECTION_RECOLLECTION_NAME/$DATA_COLLECTION_BUCKET_RECOLLECTION_NAME/random`
+    var err =
+        TestCommand.runCommandExpectExitCode(
+            1,
+            "resolve",
+            "--name="
+                + DATA_COLLECTION_RECOLLECTION_NAME
+                + "/"
+                + DATA_COLLECTION_BUCKET_RECOLLECTION_NAME
+                + "/"
+                + RandomStringUtils.random(10));
+    assertTrue(err.contains("Invalid path"));
+
+    // `terra resolve --name=$DATA_COLLECTION_RECOLLECTION_NAME/<random>`
+    var err2 =
+        TestCommand.runCommandExpectExitCode(
+            1,
+            "resolve",
+            "--name=" + DATA_COLLECTION_RECOLLECTION_NAME + "/" + RandomStringUtils.random(10));
+    assertTrue(err2.contains("Invalid path"));
+
+    // `terra resource delete --name=$name`
+    TestCommand.runCommandExpectSuccess(
+        "resource", "delete", "--name=" + DATA_COLLECTION_RECOLLECTION_NAME, "--quiet");
+  }
+
+  private void assertDataCollectionResourceResolve(JSONObject resolve) {
+    assertEquals(
+        ExternalGCSBuckets.getGsPath(DATA_COLLECTION_BUCKET_NAME),
+        resolve.get(DATA_COLLECTION_BUCKET_RECOLLECTION_NAME));
+    assertEquals(GIT_REPO_SSH_URL, resolve.get(DATA_COLLECTION_GIT_RECOLLECTION_NAME));
+    assertEquals(2, resolve.length());
+  }
+
+  private void assertDataCollectionResource(UFDataCollection actual) {
+    // For some reason resourceType is null, so don't assert resourceType.
+    assertEquals(dataCollectionUuid, actual.dataCollectionWorkspaceUuid);
+    assertEquals(DATA_COLLECTION_NAME, actual.title);
+    assertEquals(DATA_COLLECTION_DESCRIPTION, actual.shortDescription);
+    assertEquals(DATA_COLLECTION_VERSION, actual.version);
+
+    UFGcsBucket actualBucket = (UFGcsBucket) actual.resources.get(0);
+    assertEquals(DATA_COLLECTION_BUCKET_RECOLLECTION_NAME, actualBucket.name);
+    assertEquals(DATA_COLLECTION_BUCKET_NAME, actualBucket.bucketName);
```

Comment:
should we assert createdDate and updatedDate here too?

Reply:
Done, I had forgotten about that, thanks